### DISCUSSION TITLE
How to put a log in a tab

### DISCUSSION THREAD

**Discussion by Dan1001** [zauberzeug/nicegui#2009] [Q&A]
Hi,

Nicegui is great - thank you! But I've run into a small problem trying to put a log into a tab.

I've exactly followed the example [here](https://nicegui.io/documentation/log) to attach a ui.log object to the standard python logger.

I want the log to display in my "Logs" tab, so after the setup the code looks like this:

```py
with ui.tab_panel('Logs'):
        log = ui.log(max_lines=50).style('height: 512px; width: 100%;')
        logger.addHandler(LogElementHandler(log))
```

That works fine, but nicegui only starts capturing log file entries when I first click on the 'Logs' tab. All log entries prior to that point are missed. Which makes sense, given I'm initiating the log in the tab. 

How do I start the log capturing when nicegui starts? If I just move these lines to earlier in the code and include a simple 'log' in the tab then that works, but then I get the log displayed at the top of every screen. Is there a workaround?

thanks!

**Comment by falkoschindler** [comment]
Hi @Dan1001, I think this is exactly the problem we're struggling with in issue #1869.
Therefore I'll close this discussion as a duplicate. Let's continue discussing potential solutions or workarounds over there.

**Comment by Dan1001** [comment]
thank you!
